ci/common/build.bash

```diff
@@ -0,0 +1,89 @@
+#! /usr/bin/env bash
+
+# Copyright (c) 2018-2020 NVIDIA Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Released under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+
+################################################################################
+# Thrust and CUB build script for gpuCI
+################################################################################
+
+set -e
+
+# Logger function for build status output
+function logger() {
+  echo -e "\n>>>> ${@}\n"
+}
+
+################################################################################
+# VARIABLES - Set up bash and environmental variables.
+################################################################################
+
+# Get the variables the Docker container set up for us: ${CXX}, ${CUDACXX}, etc.
+source /etc/cccl.bashrc
+
+# Set path and build parallel level
+export PATH=/usr/local/cuda/bin:${PATH}
+
+# Set home to the job's workspace.
+export HOME=${WORKSPACE}
+
+# Switch to project root; also root of repo checkout.
+cd ${WORKSPACE}
+
+# If it's a nightly build, append current YYMMDD to version.
+if [ "${BUILD_MODE}" == "branch" ]; then
+  export VERSION_SUFFIX=`date +%y%m%d`
+fi
+
+# The Docker image sets up `${CXX}` and `${CUDACXX}`.
+CMAKE_FLAGS="-DCMAKE_CXX_COMPILER=${CXX} -DCMAKE_CUDA_COMPILER=${CUDACXX}"
```

Comment:
Should have quotes around each option here -- these will break if there's a space in the path to the compiler.

Reply:
Also -- this may be why you were seeing the warnings about

```
You have changed variables that require your cache to be deleted.
Configure will be re-run and you may have to reset some variables.
```

If there is an existing CMakeCache.txt in the build directory that was configured for a different compiler, CMake will get confused, give up, and start over with a clean configuration. Most options can be changed when re-invoking CMake on an existing cache, but some of the compiler specific ones are problematic.

Reply:
Quotes have been added.